elyra/pipeline/component_parser_kfp.py

```diff
@@ -140,11 +167,67 @@ def _get_adjusted_parameter_fields(self,
                 for command in component_body['implementation']['container']['command']:
                     if isinstance(command, dict) and list(command.values())[0] == io_object_name and \
                             list(command.keys())[0] == f"{io_object_type}Path":
-                        adjusted_type = "file"
+                        adjusted_type = f"{io_object_type}Path"
             if "args" in component_body['implementation']['container']:
                 for arg in component_body['implementation']['container']['args']:
                     if isinstance(arg, dict) and list(arg.values())[0] == io_object_name and \
                             list(arg.keys())[0] == f"{io_object_type}Path":
-                        adjusted_type = "file"
+                        adjusted_type = f"{io_object_type}Path"
 
         return adjusted_type
+
+    def determine_type_information(self, parsed_type: str) -> Tuple[str, str, str, bool, Any]:
+        """
+        Takes the type information of a component parameter as parsed from the component
+        specification and returns a new type that is one of several standard options.
+
+        """
```

Comment:
It looks like the difference in the returned tuple and its super class method is this one returns an extra `control` and `required_param` value.

Just curious what issues would arise if the airflow parser happened to use this same method?  Would the input/output paths be a problem?  It seems like its callers could automatically ignore the non-applicable tuples.

(We might also consider returning a dictionary of these values instead since this length of tuple is a little unwieldy - or see comment below.)